### ISSUE TITLE
Allow long-form descriptions of each metric logged.

### ISSUE THREAD

**Issue by Demonstrandum** [Demonstrandum/tensorbored#20]
While we might have a logged metric called `train/loss`, we do not have extra information about what exactly this metric is on the dashboard.

This is useful once we start logging a lot of metrics, and increasingly specific metrics, where the short-form names becomes cryptic and it becomes difficult to recall what exactly the metric means.

When hovering over a train name, we should be able to read this short description.

It's not super obvious how the description should be added, since the summary writer is usually exported by TensorFlow or PyTorch, so adding an arg there is not possible. It might have to be added to the profile, so we end up with descriptions like
```
    descriptions={'train/loss': "The loss used to optimize the model."}
```
or something...
